server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/draft/methods/MIMEMessageConverter.java

```diff
@@ -220,6 +216,46 @@ private Multipart createMultipart(CreationMessage newMessage, ImmutableList<Mess
         }
     }
 
+    private Multipart createMultipartWithAttachments(CreationMessage newMessage, ImmutableList<MessageAttachment> messageAttachments) throws IOException {
+        MultipartBuilder mixedMultipartBuilder = MultipartBuilder.create(MIXED_SUB_TYPE);
+        Map<Boolean, List<MessageAttachment>> distinguishedAttachments = distinguishAttachments(messageAttachments);
+        List<MessageAttachment> inlineAttachments = Optional.ofNullable(distinguishedAttachments.get(true))
+            .orElse(ImmutableList.of());
+        List<MessageAttachment> normalAttachments = Optional.ofNullable(distinguishedAttachments.get(false))
+            .orElse(ImmutableList.of());
+
+        if (!inlineAttachments.isEmpty()) {
+            MultipartBuilder relatedMultipart = MultipartBuilder.create(RELATED_SUB_TYPE);
+            addBody(newMessage, relatedMultipart);
+            addAttachments(inlineAttachments, relatedMultipart);
+
+            mixedMultipartBuilder
+                .addBodyPart(Message.Builder.of()
+                    .setBody(relatedMultipart.build())
+                    .build());
+        }
+
+        if (!normalAttachments.isEmpty()) {
+            addBody(newMessage, mixedMultipartBuilder);
```

Comment:
We dont add the body when we only have inlined attachments?

Reply:
We add body every time, but the body is wrapped in a multipart/related part when there are inlines